pkg/validations/validation_engine.go

```diff
@@ -97,11 +98,17 @@ func (ve *validationEngine) InitRegistry() error {
 		if check == nil {
 			return fmt.Errorf("unable to create metric for check %s", checkName)
 		}
-		metric := newGaugeVecMetric(strings.ReplaceAll(check.Spec.Name, "-", "_"),
-			// Should this be the Remediation text or the description?
-			// For now go with Description
-			check.Spec.Description,
-			[]string{"namespace", "name", "kind"})
+		ve.registeredChecks = append(ve.registeredChecks, check.Spec)
```

Comment:
If the only use for this is to provide check information by name, and since a check name has to be unique, why not may this a map?

Reply:
https://github.com/app-sre/deployment-validation-operator/blob/a3f95f2cfcf310ec98d3cb9e15c7e0c0051734ff/pkg/validations/validation_engine.go#L31-L37

https://github.com/app-sre/deployment-validation-operator/blob/a3f95f2cfcf310ec98d3cb9e15c7e0c0051734ff/pkg/validations/base.go#L52-L57

No idea why I didn't make this a map. Good idea. Updated now.